EIPS/eip-6914.md

```diff
@@ -0,0 +1,77 @@
+
+Increased state size represents load and/or complexity in client implementations. This comes in the form of client memory footprint, state root calculations, validator set scans, and more. This is a relatively simple clean-up within the state transition that will prevent the unnecessary load and complexity of the otherwise unbounded lists.
+
+## Backwards Compatibility
+
+This is a backwards incompatible change to the Consensus Layer of Ethereum and must be scheduled with a hard fork.
+
+There are no forwards/backwards compatibility issues with the Execution Layer
+
+## Test Cases
+
+Test cases are work-in-progress within the standard Consensus Layer tests.
+
+## Security Considerations
+
+Validator indices cannot be immediately reused but instead must wait `SAFE_EPOCHS_TO_REUSE_INDEX` epochs to ensure that attestations cannot be "poisoned" with withdrawn validator signatures -- thus non-slashable -- for at least the weak subjectivity period.
+
+The attestation poisoning attack hinges upon two facts:
+
+* the reuse of a validator index overwrites the previous validator's pubkey from the beacon state.
+* `AttesterSlashing`s include validator indices to reconstruct signature pariticipants.
+
+### Details of attack
+
+Assume a 1/3 attacker. Attacker exits N validators on the honest chain, where N is a small fraction of the validator set. These validators leave the exit queue and are withdrawable within a few days. Now N new deposits come in and overwrite the validators and most importantly their pubkeys.
+
+The attacker then constructs an alternative attacker chain from before any of the N voluntary exits such that the original N validators are not exited and withdrawn. N is large enough such that at least one of the N keys is on average in every committee of the attacker chain. The attacker double signs in an attempt to finalize the attacker chain but ensures that one of the N keys is mixed into any revealed double-signed aggregate attestation -- the individual attestations are unavailable, only aggregates. These malicious attestations are *not* includable in the honest chain because `AttesterSlashing`s rely upon mapping validator indices to particular pubkeys, thus breaking accountable safety.
+
+### Mitigation
+
+Not overwriting withdrawn validators for `SAFE_EPOCHS_TO_REUSE_INDEX` epochs (3x the max weak subjectivity period) ensures that attestations cannot be poisoned within the accountable safety security window.
+
+### Alternative
+
+Note that if `AttesterSlashing`s included a list of pubkeys instead of validator indices, then this would not be an issue. However this would require more breaking changes and would increase the data requirement of an `AttesterSlashing`, the largest Consensus Layer data type by a factor of 6.
```

Comment:
```suggestion
Note that if `AttesterSlashing`s included a list of pubkeys instead of validator indices, then this would not be an issue. However this would require more breaking changes and would increase the data size of `AttesterSlashing` (the largest Consensus Layer data type) by a factor of 6.
```